fedbadges/utils.py

```diff
@@ -111,13 +112,32 @@ def notification_callback(topic, msg):
 def user_exists_in_fas(config, user):
     """ Return true if the user exists in FAS. """
     default_url = 'https://admin.fedoraproject.org/accounts/'
+    if config.get["fasjson"]:
+        return user_exists_in_freeipa_fas(config, user)
+
     fas2 = fedora.client.AccountSystem(
         base_url=config['fas_credentials'].get('base_url', default_url),
         username=config['fas_credentials']['username'],
         password=config['fas_credentials']['password'],
     )
     return bool(fas2.person_by_username(user))
 
+def user_exists_in_fasjson(config, user):
+    """ Return true if the user exists in freeipa-fas. """
+    default_url = 'https://admin.fedoraproject.org/accounts/'
+    client = Client(
+        url=config['fas_credentials'].get('base_url', default_url)
+    )
+    try:
+        client.get_user(username=user).result
+        return True
+    except fasjson_client.errors.APIError as e:
+        if(e.code == 404):
+            return False
+        else:
+            log.exception(e)
+    
```

Comment:
It's good practice to always have a default bool value (here False) at the end on a boolean method, specifically when you may have from your remote host an http_code different from what you catch, here 404. If you get something else, say 400 or 308, your method will be of a NoneType and won't reflect your method signature.

Reply:
thanks @laxathom good point. I've made that change now and also made the default url global so it will only need to be changed in one place if the config isn't going to be used.